tests/integration/tests/test_gateway.py

```diff
@@ -11,6 +11,31 @@
 LOG = logging.getLogger(__name__)
 
 
+def check_gateway_service_and_port(p):
```

Comment:
From the function name I would not expect his behavior.
Looking at the invocation: `.until(lambda p: check_ingress_service_and_port(p) is not None)`, I'd expect this to return a bool. Either rename the function or change the behavior please